src/System.Data.SqlClient/src/System/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -109,6 +109,8 @@ sealed internal class SqlInternalConnectionTds : SqlInternalConnection, IDisposa
         internal SessionData _currentSessionData; // internal for use from TdsParser only, otehr should use CurrentSessionData property that will fix database and language
         private SessionData _recoverySessionData;
 
+        private static HashSet<int> transientErrors = new HashSet<int>();
```

Comment:
This could be `readonly` and should be named `s_ transientErrors`. It could also be initialized as follows, which would remove the need for the static ctor and `populateTransientErrors` method.

```c#
// The erros in the transient error set are contained in
// https://azure.microsoft.com/en-us/documentation/articles/sql-database-develop-error-messages/#transient-faults-connection-loss-and-other-temporary-errors
private static readonly HashSet<int> s_transientErrors = new HashSet<int>
{
    // SQL Error Code: 4060
    // Cannot open database "%.*ls" requested by the login. The login failed.
    4060,

    // SQL Error Code: 10928
    // Resource ID: %d. The %s limit for the database is %d and has been reached.
    10928,

    // SQL Error Code: 10929
    // Resource ID: %d. The %s minimum guarantee is %d, maximum limit is %d and the current usage for the database is %d. 
    // However, the server is currently too busy to support requests greater than %d for this database.
    10929,

    // SQL Error Code: 40197
    // You will receive this error, when the service is down due to software or hardware upgrades, hardware failures, 
    // or any other failover problems. The error code (%d) embedded within the message of error 40197 provides 
    // additional information about the kind of failure or failover that occurred. Some examples of the error codes are 
    // embedded within the message of error 40197 are 40020, 40143, 40166, and 40540.
    40197,

    40020,
    40143,
    40166,

    // The service has encountered an error processing your request. Please try again.
    40540,

    // The service is currently busy. Retry the request after 10 seconds. Incident ID: %ls. Code: %d.
    40501,

    // Database '%.*ls' on server '%.*ls' is not currently available. Please retry the connection later. 
    // If the problem persists, contact customer support, and provide them the session tracing ID of '%.*ls'.
    40613
};
```